plugins/terminal/ios.py

```diff
@@ -57,6 +59,25 @@ class TerminalModule(TerminalBase):
 
     terminal_config_prompt = re.compile(r"^.+\(config(-.*)?\)#$")
 
+    def get_privilege_level(self):
+        try:
+            cmd = {u"command": u"show privilege"}
+            result = self._exec_cli_command(
+                to_bytes(json.dumps(cmd), errors="surrogate_or_strict")
+            )
+
+            prompt = self.privilege_level_re.match(result)
+            if not prompt:
+                raise AnsibleConnectionFailure(
+                    "unable to check privilege level [%s]" % result
+                )
+
+            return int(prompt.group(1))
+        except AnsibleConnectionFailure as e:
+            raise AnsibleConnectionFailure(
+                "unable to fetch privilege, with error: %s" % (e.message)
+            )
```

Comment:
I think your exceptions might be overlapping in a way you don't want.
```suggestion
        except AnsibleConnectionFailure as e:
            raise AnsibleConnectionFailure(
                "unable to fetch privilege, with error: %s" % (e.message)
            )

        prompt = self.privilege_level_re.match(result)
        if not prompt:
            raise AnsibleConnectionFailure(
                "unable to check privilege level [%s]" % result
            )

        return int(prompt.group(1))
```